Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1835,12 +1835,17 @@ private TabularSourceData<TCellValue> ExportData<TCellValue>( DataGridExportOpti
 
         // Filter columns (exclude Command, MultiSelect, and DisplayTemplate columns)
         var columnsToExport = Columns
-                              .Where( column => column.ColumnType != DataGridColumnType.Command && column.ColumnType != DataGridColumnType.MultiSelect && column.Field != null && column.DisplayTemplate == null )
+                              .Where( column => column.ColumnType != DataGridColumnType.Command && 
+                                              column.ColumnType != DataGridColumnType.MultiSelect && 
+                                              column.Field != null && 
+                                              column.DisplayTemplate == null &&
+                                              ( options.Fields == null || options.Fields.Contains( column.Field ) ) )
                               .ToList();
-
         var exportedData = new List<List<TCellValue>>();
-
-        var columnNames = columnsToExport.Select( c => c.Caption ).ToList();
+        
+        var columnNames = (options.UseCaptions

```

Comment:
I think we should fallback to field name in case when caption is not defined?

Reply:
Is't that too much magic? 

`UseCaptionsButUseFieldIfCaptionNotSpecified`

Reply:
Not really. We only need to document `UseCaptions` by writing XML comment. And mention the behavior in the docs for example, if there is any.